src/main/java/xyz/gvital/repository/impl/RoleRepositoryImpl.java

```diff
@@ -0,0 +1,38 @@
+package xyz.gvital.repository.impl;
+
+import jakarta.persistence.EntityManagerFactory;
+import xyz.gvital.models.Role;
+import xyz.gvital.repository.RoleRepository;
+import xyz.gvital.util.JpaUtil;
+
+import java.util.Optional;
+
+public class RoleRepositoryImpl implements RoleRepository {
+
+    private final EntityManagerFactory emf = JpaUtil.getENTITY_MANAGER_FACTORY();
+
+    @Override
+    public Role create(Role role) {
+        try (var entityManager = emf.createEntityManager()) {
+            entityManager.getTransaction().begin();
+            entityManager.persist(role);
+            entityManager.getTransaction().commit();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```

Comment:
1. It is better explicitly rollback the transaction in the catch block.
2. A common approach is to throw an exception in this case, like `throw new RuntimeException("Failed to create role", e);`
3. In production, it is better to use Logger from Log4J (or Slf4J)

```suggestion
        } catch (Exception e) {
        if (transaction != null && transaction.isActive()) {
            transaction.rollback();
        }
        log.error("Failed to create Role", e);
        throw new RuntimeException("Failed to create Role", e);        }
```